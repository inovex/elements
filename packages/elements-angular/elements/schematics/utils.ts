import { virtualFs, workspaces } from '@angular-devkit/core';
import { SchematicsException, Tree } from '@angular-devkit/schematics';

const CONFIG_PATH = 'angular.json';

function createHost(tree: Tree): workspaces.WorkspaceHost {
  return {
    async readFile(path: string): Promise<string> {
      const data = tree.read(path);
      if (!data) {
        throw new SchematicsException('File not found.');
      }
      return virtualFs.fileBufferToString(data);
    },
    async writeFile(path: string, data: string): Promise<void> {
      return tree.overwrite(path, data);
    },
    async isDirectory(path: string): Promise<boolean> {
      return !tree.exists(path) && tree.getDir(path).subfiles.length > 0;
    },
    async isFile(path: string): Promise<boolean> {
      return tree.exists(path);
    },
  };
}

export function readConfig(host: Tree) {
  const sourceText = host.read(CONFIG_PATH)?.toString('utf-8') || '';
  return JSON.parse(sourceText);
}

export function writeConfig(host: Tree, config: JSON) {
  host.overwrite(CONFIG_PATH, JSON.stringify(config, null, 2));
}

export function getAngularAppConfig(
  config: any,
  projectName: string
): any | never {
  if (!config.projects.hasOwnProperty(projectName)) {
    throw new SchematicsException(`Could not find project: ${projectName}`);
  }

  return config.projects[projectName];
}

export async function getProjectName(tree: Tree) {
  const host = createHost(tree);
  const { workspace } = await workspaces.readWorkspace('/', host);
  return workspace.extensions.defaultProject as string;
}
