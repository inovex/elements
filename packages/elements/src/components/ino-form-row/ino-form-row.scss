@use '../styles/fonts';

ino-form-row {
  box-sizing: border-box;

  & > .ino-form-row__composer {
    display: flex;
    align-items: center;
    box-sizing: border-box;
    width: 100%;
    padding: 2px;
  }

  .ino-form-row__element {
    flex: 1;
  }

  .ino-form-row__label {
    width: 120px;
  }
}
