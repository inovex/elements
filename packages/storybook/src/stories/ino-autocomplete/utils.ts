import { KeyValue } from '@inovex.de/elements';

export const AutoCompleteStoryOptions: KeyValue[] = [
  {
    key: 'hh',
    value: 'Hamburg',
  },
  {
    key: 'ber',
    value: 'Berlin',
  },
  {
    key: 'k',
    value: 'Karlsruhe',
  },
];
