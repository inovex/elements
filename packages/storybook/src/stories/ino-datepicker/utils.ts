import '@inovex.de/elements';

export function registerInlineDatepickerHandler() {
  const group = document.querySelector('ino-radio-group') as HTMLInoRadioGroupElement;
  const datepicker = document.querySelector('.datepicker-group ino-datepicker') as HTMLInoDatepickerElement;

  const today = document.querySelector('.today');
  const thisWeek = document.querySelector('.thisWeek');
  const lastWeek = document.querySelector('.lastWeek');

  const todayClickHandler = () => {
    datepicker.value = '02.03.2020';
    datepicker.label = 'Am';
    group.value = 'at';
    datepicker.range = false;
  };

  const thisWeekClickHandler = () => {
    group.value = 'range';
    datepicker.label = 'Zeitraum';
    datepicker.range = true;
    datepicker.value = '02.03.2020 bis 08.03.2020';
  };

  const lastWeekClickHandler = () => {
    group.value = 'range';
    datepicker.label = 'Zeitraum';
    datepicker.range = true;
    datepicker.value = '24.02.2020 bis 01.03.2020';
  };

  const checkedChangeHandler = (e: CustomEvent<boolean>) => {
    const target = e.target as HTMLInoRadioElement;

    switch (target.value) {
      case 'at':
        datepicker.label = 'Am';
        datepicker.range = false;
        break;
      case 'after':
        datepicker.label = 'Ab';
        datepicker.range = false;
        if (group.value == 'range') {
          datepicker.value = datepicker.value.split(' - ').shift();
        }
        break;
      case 'before':
        datepicker.label = 'Bis';
        datepicker.range = false;
        if (group.value == 'range') {
          datepicker.value = datepicker.value.split(' - ').pop();
        }
        break;
      case 'range':
        datepicker.range = true;
        datepicker.label = 'Zeitraum';
        break;
    }
    group.value = target.value;
  };

  today.addEventListener('click', todayClickHandler);
  thisWeek.addEventListener('click', thisWeekClickHandler);
  lastWeek.addEventListener('click', lastWeekClickHandler);
  group.addEventListener('checkedChange', checkedChangeHandler);

  return () => {
    today.removeEventListener('click', todayClickHandler);
    thisWeek.removeEventListener('click', thisWeekClickHandler);
    lastWeek.removeEventListener('click', lastWeekClickHandler);
    group.removeEventListener('checkedChange', checkedChangeHandler);
  };
}
