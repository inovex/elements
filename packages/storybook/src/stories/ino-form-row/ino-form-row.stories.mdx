import { Meta, Description } from "@storybook/addon-docs";
import readme from "../../../../elements/src/components/ino-form-row/readme.md";

<Meta title="Input/ino-form-row/Documentation" />

<Description markdown={readme} />
