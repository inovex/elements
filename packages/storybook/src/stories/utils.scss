/**
 * Applies styling to the container of a story.
 * The styling is encapsulated within this mixin because the ID may change in the future.
 */
@mixin story-container {
  #root-inner {
    @content;
  }
}
