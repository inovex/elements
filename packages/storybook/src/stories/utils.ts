import { html } from 'lit-html';
import ICONS from '../../../elements/src/components/ino-icon/icons';

export const maybeCreateStoryArgs = (story: any) => {
  if (!story.args) {
    story.args = {};
  }
  if (!story.argTypes) {
    story.argTypes = {};
  }
};

export const withIconControl = (
  story: any,
  propertyName: string,
  defaultValue?: string
) => {
  maybeCreateStoryArgs(story);
  story.args[propertyName] = defaultValue || '';
  story.argTypes[propertyName] = {
    control: {
      type: 'select',
    },
    options: getIcons(),
  };
};

export const withColorScheme = (
  story: any,
  propertyName: string,
  defaultValue?: string
) => {
  maybeCreateStoryArgs(story);
  story.args[propertyName] = defaultValue || '';
  story.argTypes[propertyName] = {
    control: {
      type: 'select',
    },
    options: [
      'primary',
      'secondary',
      'success',
      'warning',
      'error',
      'light',
      'dark',
    ],
  };
};

export const withSortDirection = (
  story: any,
  propertyName: string,
  defaultValue?: string
) => {
  maybeCreateStoryArgs(story);
  story.args[propertyName] = defaultValue || '';
  story.argTypes[propertyName] = {
    options: ['desc', 'asc'],
    control: {
      type: 'select',
    },
  };
};

export const getIcons = () => ICONS.filter((icon) => !icon.startsWith('_'));

export const defaultDecorator = (story, className?: string) => {
  return html` <div class="${className ?? ''}">${story()}</div> `;
};
