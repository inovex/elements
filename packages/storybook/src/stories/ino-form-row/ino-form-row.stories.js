import { withActions } from '@storybook/addon-actions';
import addons from '@storybook/addons';
import CoreEvents from '@storybook/core-events';

import { boolean, text } from '@storybook/addon-knobs';

import withStencilReadme from '_local-storybookcore/with-stencil-readme';

import componentReadme from '_local-elements/src/components/ino-form-row/readme.md';
import './ino-form-row.scss';

// https://github.com/storybooks/storybook/issues/4337#issuecomment-428495664
function subscribeToComponentEvents() {
  // == event block
  const eventHandler = function (e) {
    const el = e.target;
    if (el.tagName.toLowerCase() !== 'ino-input') {
      return;
    }

    e.target.setAttribute('value', e.detail);
  };

  document.addEventListener('valueChange', eventHandler);
  // == event block

  // unsubscribe function will be called by Storybook
  return () => {
    document.removeEventListener('valueChange', eventHandler);
  };
}

export default {
  title: 'Input/<ino-form-row>',
  parameters: {
    actions: {
      handles: ['valueChange .customizable-form-row']
    }
  },
  decorators: [
    withStencilReadme(componentReadme),
    (story) => {
      addons.getChannel().emit(CoreEvents.REGISTER_SUBSCRIPTION, subscribeToComponentEvents);
      return story();
    },
  ],
};

export const DefaultUsage = () => /*html*/ `
    <div class="story-form-row">
      <div>
        <h4>Customizable Form Row</h4>
        <ino-form-row
          class="customizable-form-row"
          ino-label="${text('ino-label', 'inovex-elements')}"
          ino-mandatory="${boolean('ino-mandatory', true)}"
        >
          <ino-input></ino-input>
        </ino-form-row>
      </div>
    </div>
  `;

DefaultUsage.storyName = 'Default usage';
