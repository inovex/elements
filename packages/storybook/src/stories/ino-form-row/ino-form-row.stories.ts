import { Components } from '@inovex.de/elements';
import { useEffect } from '@storybook/client-api';
import { Story } from '@storybook/web-components';
import { html } from 'lit-html';
import { defaultDecorator } from '../utils';

import './ino-form-row.scss';

export default {
  title: 'Input/ino-form-row',
  component: 'ino-form-row',
  parameters: {
    actions: {
      handles: ['valueChange .customizable-form-row'],
    },
  },
  decorators: [
    defaultDecorator,
    (story) => {
      useEffect(() => {
        const valueChangeHandler = function (e) {
          if (e.target.tagName.toLowerCase() !== 'ino-input') {
            return;
          }

          const input = e.target as HTMLInoInputElement;
          input.value = e.detail;
        };

        document.addEventListener('valueChange', valueChangeHandler);
        // == event block

        // unsubscribe function will be called by Storybook
        return () => {
          document.removeEventListener('valueChange', valueChangeHandler);
        };
      });
      return story();
    },
  ],
};

export const Playground: Story<Components.InoFormRow> = (args) => html`
  <ino-form-row
    class="customizable-form-row"
    label="${args.label}"
    mandatory="${args.mandatory}"
  >
    <ino-input></ino-input>
  </ino-form-row>
`;

Playground.args = {
  label: 'Label',
  mandatory: false,
};
