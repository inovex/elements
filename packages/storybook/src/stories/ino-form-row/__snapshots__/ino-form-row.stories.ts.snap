// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input/ino-form-row Mandatory smoke-test 1`] = `
<div id="root-inner">
  <div class>
    <ino-form-row class="customizable-form-row hydrated"
                  label="Label"
                  mandatory="true"
    >
      <div class="ino-form-row__composer">
        <div class="ino-form-row__label">
          Label
          <span>
            *
          </span>
        </div>
        <div class="ino-form-row__element">
          <ino-input class="hydrated"
                     tabindex="-1"
          >
            <label class="ino-input__composer mdc-text-field mdc-text-field--filled mdc-text-field--box mdc-text-field--no-label mdc-ripple-upgraded">
              <ino-label class="hydrated">
                <div class="mdc-line-ripple">
                </div>
              </ino-label>
              <input class="mdc-text-field__input"
                     step="1"
                     type="text"
              >
            </label>
            <div class="mdc-text-field-helper-line">
            </div>
          </ino-input>
        </div>
      </div>
    </ino-form-row>
  </div>
</div>
`;

exports[`Input/ino-form-row Playground smoke-test 1`] = `
<div id="root-inner">
  <div class>
    <ino-form-row class="customizable-form-row hydrated"
                  label="Label"
                  mandatory="false"
    >
      <div class="ino-form-row__composer">
        <div class="ino-form-row__label">
          Label
        </div>
        <div class="ino-form-row__element">
          <ino-input class="hydrated"
                     tabindex="-1"
          >
            <label class="ino-input__composer mdc-text-field mdc-text-field--filled mdc-text-field--box mdc-text-field--no-label mdc-ripple-upgraded">
              <ino-label class="hydrated">
                <div class="mdc-line-ripple">
                </div>
              </ino-label>
              <input class="mdc-text-field__input"
                     step="1"
                     type="text"
              >
            </label>
            <div class="mdc-text-field-helper-line">
            </div>
          </ino-input>
        </div>
      </div>
    </ino-form-row>
  </div>
</div>
`;
